src/pipeline/rackhd/npm/Publish.groovy

```diff
@@ -0,0 +1,92 @@
+package pipeline.rackhd.npm
+
+
+//=============================================================================================
+// Publish NPM packages according to manifest file , to remote NPM registry ( it will clone code first, and run npm-publish)
+//
+// Inputs:
+//  manifest_file - the manifest file contains all repo information(URL/commit/branch) which to be published, can be a local file or remote URL
+//  is_offical_release  - the flag to tell if publishing a offical release or a dev build. the version naming will vary.
+//  npm_registry  - the URL of the npm registry
+//  npm_registry_token - the upload credential token of the npm registry
+//  on_build_config_dir - the on-build-config dir where the python scripts come from. the on-build-config src code can be prepared before using this function, or this function will download latest src for you.)(by default: $WORKSPACE/on-build-config)
+//  clone_dest_dir      - where the code will be cloned to, by default it is blank, and will use a temporary folder under /tmp to clone code.
+//
+// ===============================================================================================
+//
+//  Example Usage:
+/
+//  node
+//    {
+//        def manifest_file="https://dl.bintray.com/rackhd/binary/master-20170827"
+//        def is_offical_release=false;
+//
+//        def npm_publisher = new pipeline.rackhd.npm.Publish()
+//        withCredentials([
+//                usernamePassword(credentialsId: '736849f6-ba2c-489d-b5ca-d1b1f4be2252', 
+//                passwordVariable: 'NPM_TOKEN', 
+//                usernameVariable: 'NPM_REGISTRY')])
+//        {
+//              npm_publisher.publish(manifest_file, is_offical_release, NPM_TOKEN, NPM_REGISTRY, "$WORKSPACE/on-build-config", "/tmp/clone_dir" )
+//        }
+//    }
+//=============================================================================================
+def publish(String manifest_file, Boolean is_offical_release, String  npm_registry , String npm_registry_token , String on_build_config_dir = "$WORKSPACE/on-build-config", String clone_dest_dir="" )
+{
+    deleteDir()
```

Comment:
Please remove this line,
It may remove something required by other steps.

Reply:
fixed. thank you